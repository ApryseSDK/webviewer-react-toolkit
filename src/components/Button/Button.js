'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Button = void 0;
var tslib_1 = require('tslib');
var classnames_1 = tslib_1.__importDefault(require('classnames'));
var react_1 = tslib_1.__importStar(require('react'));
var hooks_1 = require('../../hooks');
exports.Button = (0, react_1.forwardRef)(function(_a, ref) {
  var _b = _a.buttonStyle,
    buttonStyle = _b === void 0 ? 'default' : _b,
    _c = _a.buttonSize,
    buttonSize = _c === void 0 ? 'default' : _c,
    _d = _a.type,
    type = _d === void 0 ? 'button' : _d,
    className = _a.className,
    children = _a.children,
    buttonProps = tslib_1.__rest(_a, [
      'buttonStyle',
      'buttonSize',
      'type',
      'className',
      'children',
    ]);
  var isUserTabbing = (0, hooks_1.useAccessibleFocus)();
  var buttonClass = (0, classnames_1.default)(
    'ui__base ui__button',
    'ui__button--style-'.concat(buttonStyle),
    'ui__button--size-'.concat(buttonSize),
    {
      'ui__button--disabled': buttonProps.disabled,
      'ui__button--tabbing': isUserTabbing,
    },
    className,
  );
  return react_1.default.createElement(
    'button',
    tslib_1.__assign({}, buttonProps, {
      className: buttonClass,
      type: type,
      ref: ref,
    }),
    react_1.default.createElement(
      'div',
      { className: 'ui__button__internal' },
      children,
    ),
  );
});
//# sourceMappingURL=Button.js.map
