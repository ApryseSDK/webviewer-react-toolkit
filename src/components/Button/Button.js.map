{"version":3,"file":"Button.js","sourceRoot":"","sources":["Button.tsx"],"names":[],"mappings":";;;;AAAA,kEAAoC;AACpC,qDAAgE;AAChE,qCAAiD;AAoBpC,QAAA,MAAM,GAAG,IAAA,kBAAU,EAC9B,UAAC,EAAyG,EAAE,GAAG;IAA5G,IAAA,mBAAuB,EAAvB,WAAW,mBAAG,SAAS,KAAA,EAAE,kBAAsB,EAAtB,UAAU,mBAAG,SAAS,KAAA,EAAE,YAAe,EAAf,IAAI,mBAAG,QAAQ,KAAA,EAAE,SAAS,eAAA,EAAE,QAAQ,cAAA,EAAK,WAAW,sBAAvG,8DAAyG,CAAF;IACtG,IAAM,aAAa,GAAG,IAAA,0BAAkB,GAAE,CAAC;IAE3C,IAAM,WAAW,GAAG,IAAA,oBAAU,EAC5B,qBAAqB,EACrB,4BAAqB,WAAW,CAAE,EAClC,2BAAoB,UAAU,CAAE,EAChC;QACE,sBAAsB,EAAE,WAAW,CAAC,QAAQ;QAC5C,qBAAqB,EAAE,aAAa;KACrC,EACD,SAAS,CACV,CAAC;IAEF,OAAO,CACL,6DAAY,WAAW,IAAE,SAAS,EAAE,WAAW,EAAE,IAAI,EAAE,IAAI,EAAE,GAAG,EAAE,GAAG;QACnE,uCAAK,SAAS,EAAC,sBAAsB,IAAE,QAAQ,CAAO,CAC/C,CACV,CAAC;AACJ,CAAC,CACF,CAAC"}