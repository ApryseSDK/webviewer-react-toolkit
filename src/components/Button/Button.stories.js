'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Basic = void 0;
var tslib_1 = require('tslib');
var addon_knobs_1 = require('@storybook/addon-knobs');
var react_1 = tslib_1.__importDefault(require('react'));
var action_1 = require('../../storybook-helpers/action/action');
var Button_1 = require('../Button');
var README_md_1 = tslib_1.__importDefault(require('./README.md'));
exports.default = {
  title: 'Components/Button',
  component: Button_1.Button,
  parameters: { readme: README_md_1.default },
};
var Basic = function() {
  return react_1.default.createElement(
    Button_1.Button,
    {
      buttonStyle: (0, addon_knobs_1.select)(
        'buttonStyle',
        ['default', 'borderless', 'outline'],
        'default',
      ),
      buttonSize: (0, addon_knobs_1.select)(
        'buttonSize',
        ['small', 'default', 'large'],
        'default',
      ),
      disabled: (0, addon_knobs_1.boolean)('disabled', false),
      onClick: (0, action_1.action)('onClick'),
    },
    (0, addon_knobs_1.text)('children', 'Button content'),
  );
};
exports.Basic = Basic;
//# sourceMappingURL=Button.stories.js.map
