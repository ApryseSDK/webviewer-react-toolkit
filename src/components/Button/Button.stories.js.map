{"version":3,"file":"Button.stories.js","sourceRoot":"","sources":["Button.stories.tsx"],"names":[],"mappings":";;;;AAAA,sDAA+D;AAC/D,wDAA0B;AAC1B,gEAA+D;AAC/D,oCAAmC;AACnC,kEAAiC;AAEjC,kBAAe,EAAE,KAAK,EAAE,mBAAmB,EAAE,SAAS,EAAE,eAAM,EAAE,UAAU,EAAE,EAAE,MAAM,qBAAA,EAAE,EAAE,CAAC;AAElF,IAAM,KAAK,GAAG,cAAM,OAAA,CACzB,8BAAC,eAAM,IACL,WAAW,EAAE,IAAA,oBAAM,EAAC,aAAa,EAAE,CAAC,SAAS,EAAE,YAAY,EAAE,SAAS,CAAC,EAAE,SAAS,CAAC,EACnF,UAAU,EAAE,IAAA,oBAAM,EAAC,YAAY,EAAE,CAAC,OAAO,EAAE,SAAS,EAAE,OAAO,CAAC,EAAE,SAAS,CAAC,EAC1E,QAAQ,EAAE,IAAA,qBAAO,EAAC,UAAU,EAAE,KAAK,CAAC,EACpC,OAAO,EAAE,IAAA,eAAM,EAAC,SAAS,CAAC,IAEzB,IAAA,kBAAI,EAAC,UAAU,EAAE,gBAAgB,CAAC,CAC5B,CACV,EAT0B,CAS1B,CAAC;AATW,QAAA,KAAK,SAShB"}