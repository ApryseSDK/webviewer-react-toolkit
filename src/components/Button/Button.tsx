import classnames from 'classnames';
import React, { forwardRef, ButtonHTMLAttributes } from 'react';

export enum ButtonStyle {
  Default = 'default',
  Borderless = 'borderless',
  Outline = 'outline',
}

export enum ButtonSize {
  Small = 'small',
  Default = 'default',
  Large = 'large',
}

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  /**
   * Set the visual appearance of the button.
   * @default ButtonStyle.Default
   */
  buttonStyle?: ButtonStyle;
  /**
   * Set the visual appearance of the button.
   * @default ButtonSize.Default
   */
  buttonSize?: ButtonSize;
  /**
   * Defaults to 'button' instead of 'submit' to prevent accidental submissions.
   * @default "button"
   */
  type?: ButtonHTMLAttributes<HTMLButtonElement>['type'];
}

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  (
    { buttonStyle = ButtonStyle.Default, buttonSize = ButtonSize.Default, type = 'button', className, ...buttonProps },
    ref,
  ) => {
    const buttonClass = classnames(
      'ui__base ui__button',
      `ui__button--style-${buttonStyle}`,
      `ui__button--size-${buttonSize}`,
      { ['ui__button--disabled']: buttonProps.disabled },
      className,
    );

    return <button {...buttonProps} className={buttonClass} type={type} ref={ref} />;
  },
);
