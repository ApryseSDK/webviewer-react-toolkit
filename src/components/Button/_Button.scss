.ui__button {
  cursor: pointer;
  outline: none;
  user-select: none;

  &--disabled {
    //
  }

  &:focus {
    @include focus;
  }

  /* --- Button styles. --- */

  &--style {
    &-default {
      border-color: $blue-gray-5;
      background-color: $blue-gray-5;
      color: white;
    }

    &-borderless {
      border-color: transparent;
      background-color: transparent;
      color: $blue-gray-5;
    }

    &-outline {
      border-color: $blue-gray-5;
      background-color: transparent;
      color: $blue-gray-5;
    }
  }

  &--size {
    &-small {
      border-radius: $border-radius-small;
    }

    &-default {
      border-radius: $border-radius;
      padding: $padding-tiny $padding-small;
    }

    &-large {
      border-radius: $border-radius;
      padding: $padding-small $padding;
    }
  }
}
