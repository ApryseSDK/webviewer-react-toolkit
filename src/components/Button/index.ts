export { Button as default, ButtonProps } from './Button';
