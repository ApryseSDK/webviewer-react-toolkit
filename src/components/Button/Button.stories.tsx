import { action } from '@storybook/addon-actions';
import { boolean, select, text } from '@storybook/addon-knobs';
import React from 'react';
import { Button } from '../Button';
import info from './README.md';

export default { title: 'Components/Button', component: Button, parameters: { info } };

export const Basic = () => (
  <Button
    buttonStyle={select('buttonStyle', ['default', 'borderless', 'outline'], 'default')}
    buttonSize={select('buttonSize', ['small', 'default', 'large'], 'default')}
    disabled={boolean('disabled', false)}
    onClick={action('onClick')}
  >
    {text('children', 'Button content')}
  </Button>
);
