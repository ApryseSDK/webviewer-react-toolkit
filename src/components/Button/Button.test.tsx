import { mount, shallow } from 'enzyme';
import React from 'react';
import { spy } from 'sinon';
import { Button } from '../Button';

describe('Button component', () => {
  it('renders its contents', () => {
    const button = shallow(<Button />);
    expect(button.find('.ui__button')).toHaveLength(1);
  });

  it('snapshot renders default button', () => {
    const button = shallow(<Button />);
    expect(button).toMatchSnapshot();
  });

  it('clicking button triggers onClick prop', () => {
    const onClick = spy();
    shallow(<Button onClick={onClick} />).simulate('click');
    expect(onClick.callCount).toBe(1);
  });

  it('clicking disabled button does not trigger onClick prop', () => {
    const onClick = spy();
    mount(<Button onClick={onClick} disabled />).simulate('click');
    expect(onClick.callCount).toBe(0);
  });
});
