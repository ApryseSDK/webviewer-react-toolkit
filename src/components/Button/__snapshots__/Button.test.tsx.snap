// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot renders default button 1`] = `
<button
  className="ui__base ui__button ui__button--default"
  type="button"
/>
`;
